src/main/java/Listener/CountingEnforcement.java

```diff
@@ -39,25 +43,38 @@ private void countCheck(TextChannel channel, Message message, MessageAuthor auth
             return;
         }
 
-        Integer input;
-        User user;
-        if (channel.getMessages(2).get().getOldestMessage().get().getAuthor().getIdAsString().equals(BotInfo.getBotId())) {
-            input = Integer.parseInt(channel.getMessages(3).get().getOldestMessage().get().getContent().replaceAll("\\*", "")) + 1;
-            user = channel.getMessages(3).get().getOldestMessage().get().getAuthor().asUser().get();
+        Integer previousNum;
+        User previousUser;
+        MessageSet lastTwoMessages = channel.getMessages(2).get();
+        Boolean previousMessageAuthorIsTheBot = lastTwoMessages.getOldestMessage().get().getAuthor().getIdAsString().equals(BotInfo.getBotId());
+        Message previousHumanMessage;
+        if (previousMessageAuthorIsTheBot) {
+            // assumes the econd to last message is not written by the bot implicitly
+            // there's probably a good pattern for scrolling up the last few message to find a non-bot message
+            // but i'm getting bored
```

Comment:
https://docs.javacord.org/api/build/5110/org/javacord/api/entity/message/MessageSet.html says

> This class represents an unmodifiable set of messages that is always sorted from oldest to newest according to the natural ordering of Messages.

So the set is already ordered, you could probably get the last 5 messages and then loop over the first 4 assigning each non-bot one to `previousHumanMessage` and then you've more effectively mitigated this issue and guaranteed you have the most recent message from a human.

Reply:
Dammit, now i might implement that